os/arch/arm/src/armv7-a/iar/arm_fetchadd.S

```diff
@@ -0,0 +1,223 @@
+/****************************************************************************
```

Comment:
We do not support IAR. 
So, let us remove this folder. And move the files from GNU folder outside to the v7a folder.